### DISCUSSION TITLE
Guess the Number Game

### DISCUSSION THREAD

**Discussion by Capital-EX** [effekt-lang/effekt#259] [Show and tell]
Hi, I wanted to share a guess the number game I made with Effekt. It's the first time I've experimented with algebraic effects so I was just placing them wherever I could squeeze them.

https://codeberg.org/CapitalEx/effekt-guessing-game/src/branch/main

I did find on bug with the `chez` backends. Running `(get-line (current-input-port))` would freeze the repl. Terminating the session using `ctrl-c` would cause the process to swiftly consume memory.

**Comment by b-studios** [comment]
Hey, welcome! And thanks for reporting this bug. Have you tried running the example outside of the effekt repl / sbt? I also noticed problems with the chez backend, when starting the program from within `sbt` / the Java .jar

The scheme file will be generated in the `out` folder and you should be able to start it directly with `scheme --script out/myfile.ss` (https://github.com/effekt-lang/effekt/blob/master/effekt/jvm/src/main/scala/effekt/Runner.scala#L116)

**Reply by Capital-EX** [reply]
It works when running outside of the effekt repl. That's what my small shell script does:
```sh
#!/bin/env sh
effekt.sh --backend chez-callcc -c main.effekt
scheme --script out/main.ss
```

**Reply by b-studios** [reply]
Thanks for confirming and sorry for the inconvenience. I am glad it works for you .
